modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLValidatorImpl.java

```diff
@@ -130,20 +130,31 @@ public void validateDirectoryName(String directoryName)
 	public void validateFileExtension(String fileName)
 		throws FileExtensionException {
 
+		boolean restrictAllFileExtensions = false;
 		boolean validFileExtension = false;
 
 		String[] fileExtensions = _dlConfiguration.fileExtensions();
 
 		for (String fileExtension : fileExtensions) {
-			if (StringPool.STAR.equals(fileExtension) ||
-				StringUtil.endsWith(fileName, fileExtension)) {
-
-				validFileExtension = true;
+			if (fileExtension.isEmpty()) {
```

Comment:
I don't fully understand why we need this change of logic here. The way I see it, the validation logic is correct and we only need to take care of the exception thrown in the UI side. 
Could you please explain why we need to change the way we validate the extension? Is there any bug here?
Looking at the logic it seems that we are changing the validation now so in case there's an empty extension we should not allow to add any extension at all. I don't think we should support that use case and we should actually treat that as a configuration bug. What is the point of not allowing any extension to be uploaded? That leaves the documents and media unusable right?

Reply:
Hi @sergiogonzalez,

Yes, I can give more details to you!

The previous logic, without my modifications, is considering an empty extension as a valid file extension, because the result of the function **StringUtil.endsWith(fileName, fileExtension)** is **true**. Hence, an empty extension is representing the same as input an * in the file extensions configuration and this current behavior is different from the one on 7.0.x. On 7.0.x, an empty extension isn't considered a valid file extension and what is causing this behavior difference is the way the array **fileExtensions** is being populated on these both branches, master and 7.0.x. On master, the array is populated by the [_dlConfiguration.fileExtensions](https://github.com/liferay/liferay-portal/blob/6bbbd0e0965124098d00761f7328ecf308caf355/modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLValidatorImpl.java#L135) method and, on 7.0.x, the array is populated by the [PrefsPropsUtil.getStringArray(PropsKeys.DL_FILE_EXTENSIONS, StringPool.COMMA)](https://github.com/liferay/liferay-portal-ee/blob/6048ceab15ff47de4c8731c7eb577511d6df6293/modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLValidatorImpl.java#L115-L116), this difference generates, respectively, an empty array and an array with one element, which is the empty extension.

Therefore, this logic intends to keep the behavior available on 7.0.x. In addition, as you've already mentioned, what is the purpose of a Documents and Media field that no file can be uploaded? This logic helps to address this use case with a more specific error message, which will instruct the user how to fix this misconfiguration.

If you need any other information, please let me know.

Thanks.

Reply:
Hey @natocesarrego 
Thanks for the additional information. It seems that is a different issue and it should be treated differently in a new LPS.
Could you please create an LPS ticket explaining the issue (basically what you described above) so QA is able to reproduce the issue and then send a separate pull request with a separate LPS for that fix?
Thanks!

Reply:
Yeah, I can do it, @sergiogonzalez. I'll split this complete solution in two different pulls.

Thanks.